debian/mariadb-server.mariadb.init

```diff
@@ -19,9 +19,10 @@ ${DEBIAN_SCRIPT_DEBUG:+ set -v -x}
 
 test -x /usr/sbin/mariadbd || exit 0
 
+# shellcheck source=/dev/null
 . /lib/lsb/init-functions
 
-SELF=$(cd "$(dirname $0)"; pwd -P)/$(basename $0)
+SELF="$(cd "$(dirname "$0")"; pwd -P)/$(basename "$0")"
```

Comment:
This quoting here seems fishy.

The inner quotes should be escaped. For e.g.:
```
"$(cd \"$(dirname \"$0\")\"; pwd -P)/$(basename \"$0\")"
```

Reply:
No this is not necessary, but the correct thing to do here would be to use parameter expansion since we are using bash:

Something like the following (not tested):
```bash
SELF="$(cd "${0%/*}"; pwd -P)/${0##*/}"
```

Reply:
Ok I've tested both:
```
echo "${0%/*}"
echo "$(dirname "${0}")"
echo "$(dirname \"${0}\")"
```
An then run it with `bash test.sh` which gives:
```
test.sh
.
.
```
So @fauust approach is correct when there is directory name but fails when run in same dir. Also original works as expected. This is very difficult way to find `realpath`-command?